snap/quota/resources.go

```diff
@@ -262,6 +311,27 @@ func (qr *Resources) ValidateChange(newLimits Resources) error {
 		}
 	}
 
+	// Verify journal limits not being removed
+	if qr.Journal != nil && newLimits.Journal != nil {
+		if qr.Journal.Size != nil && newLimits.Journal.Size != nil && newLimits.Journal.Size.Limit == 0 {
+			return fmt.Errorf("cannot remove journal size limit from quota group")
+		}
+
+		// The lower limit for the journal size is arbitrarily set to 64kb to protect against any
+		// accidental values that are very low.
+		if newLimits.Journal.Size != nil && newLimits.Journal.Size.Limit < 64*quantity.SizeKiB {
+			return fmt.Errorf("journal size limit %d is too small: size must be larger than 64KB", newLimits.Journal.Size.Limit)
```

Comment:
Why isn't this limit enforced in validateJournalQuota, along 4GB limit check?

Reply:
Good question, we have two types of validation;

.Validate() which is called as a part of routine validation (i.e when creating/modifying/retrieving) groups, which checks for values being zero primarily, but the 4GB limit check can also be here as the 4GB limit is currently fixed in systemd so lets catch that if it were modified in the state.json

.ValidateChange() which is only called when creating or modifying groups, and will supply new limits against current limits, and here we do some additional validation, including specific limit sizes - like memory limit being low, and now the journal size, we keep these checks here in case we want to modify it in the future and don't want to break existing groups that violate the new limits.